src/Symfony/Component/Runtime/SymfonyRuntime.php

```diff
@@ -0,0 +1,208 @@
+            }
+
+            (new Dotenv())->bootEnv(\is_string($path) ? $options['dotenv'] : (\dirname(__DIR__, 3).'/.env'));
+        }
+
+        if (!class_exists(ErrorHandler::class)) {
+            throw new \LogicException(sprintf('You cannot use "%s" as the ErrorHandler component is not installed. Try running "composer require symfony/error-handler".', __CLASS__));
+        }
+
+        if ($_SERVER['APP_DEBUG']) {
+            umask(0000);
+            Debug::enable();
+        } else {
+            ErrorHandler::register();
+        }
+    }
+
+    public function getApp(\Closure $app): \Closure
+    {
+        $arguments = [];
+
+        try {
+            foreach ((new \ReflectionFunction($app))->getParameters() as $parameter) {
+                $arguments[] = $this->getArgument($parameter);
+            }
+        } catch (\InvalidArgumentException $e) {
+            if (!$parameter->isOptional()) {
+                throw $e;
+            }
+        }
+
+        if (!$arguments) {
+            return $app;
+        }
+
+        return static function () use ($app, $arguments) {
+            return $app(...$arguments);
+        };
+    }
+
+    public function getMain(object $app): \Closure
+    {
+        if ($app instanceof HttpKernelInterface) {
+            $request = $this->request ?? $this->request = Request::createFromGlobals();
+
+            return static function () use ($app, $request): int {
+                $response = $app->handle($request);
+                $response->send();
+
+                if ($app instanceof TerminableInterface) {
+                    $app->terminate($request, $response);
+                }
+
+                return 0;
+            };
+        }
+
+        if ($app instanceof Response) {
+            return static function () use ($app): int {
+                $app->send();
+
+                return 0;
+            };
+        }
+
+        if ($app instanceof Command) {
+            $application = $this->application ?? $this->application = new Application();
+            $main = $this->getMain($application);
+
+            return static function () use ($app, $application, $main): int {
+                $application->setName($app->getName() ?: $application->getName());
+
+                if (!$app->getName() || !$application->has($app->getName())) {
+                    $app->setName($_SERVER['argv'][0]);
+                    $application->add($app);
+                }
+
+                $application->setDefaultCommand($app->getName(), true);
+
+                return $main();
+            };
+        }
+
+        if ($app instanceof Application) {
+            if (!\in_array(\PHP_SAPI, ['cli', 'phpdbg', 'embed'], true)) {
+                echo 'Warning: The console should be invoked via the CLI version of PHP, not the '.\PHP_SAPI.' SAPI'.\PHP_EOL;
+            }
+
+            set_time_limit(0);
```

Comment:
shouldnt this be kept inside the return closure? same for the global echo above

Reply:
I thought about this and my conclusion is that doing this outside is more flexible: a decorator can decide to hook after and reconfigure. Moving inside would prevent this.

This led me to the following conclusion: the very first line of the returned closure should probably always use the `$app` variable immediately. Any logic before should be moved outside, for the reason I just gave.

Reply:
it's still a global side effect :( same for everything __construct is doing. Isnt the component meant to serve an isolated runtime? I feel like it breaks its own design rules.

should there be some setup/teardown extension points?

Reply:
> Isnt the component meant to serve an isolated runtime

the app is totally decoupled from global state, and the very purpose of a runtime instance *is precisely* to deal with the global state, so that goal is achieved.

> should there be some setup/teardown extension points?

there already is one: the high-order interface allows doing anything you want. Tell me if you think of a case where this is not the case.

Reply:
you're right, RuntimeInterface can do anything it needs to do. As it represents the actual runtime :}